app/services/declarations/create.rb

```diff
@@ -30,6 +32,15 @@ class Create
     validate :validates_billable_slot_available
     validate :declaration_date_not_in_the_future
 
+    validates :delivery_partner_id, presence: true, if: -> { Feature.declarations_require_delivery_partner? }
```

Comment:
This rule is a bit more complicated now in the Declaration model and is going to get more complicated (we need to handle overseas applications differently it turns out)